### PR TITLE
DCOS-45657: added Icon component

### PR DESCRIPTION
**Pull request by mperrotti**
Added an Icon component and previews for our "system" and "product" icon sets.

The "default" icon story has Storybook knobs so you can preview different shapes, colors, and sizes

### CODE REVIEW

**Comment by mperrotti** [comment]
This PR contains some of the changes in https://github.com/dcos-labs/ui-kit/pull/272. I was initially waiting to push this up, but I want to try and make sure these both get closed out this sprint.

**This means you can ignore the changes in `packages/icons/`**

File: packages/icon/style.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { css } from "emotion";
+
+// position: relative;
+// vertical-align: bottom;
```
Comment (GeorgiSTodorov) [inline packages/icon/style.ts]: Are these left here on purpose?

File: packages/icon/components/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import { ProductIcons } from "../../icons/dist/product-icons-enum";
+import { icon } from "../style";
+
+export enum IconSizes {
+  XS = iconSizeXs as any,
+  S = iconSizeS as any,
+  M = iconSizeM as any,
+  L = iconSizeL as any,
+  XL = iconSizeXl as any,
+  XXL = iconSizeXxl as any
```
Comment (GeorgiSTodorov) [inline packages/icon/components/Icon.tsx]: Is having `as any` here really necessary?

File: packages/icon/style.ts (hunk 1 above)
Comment (mperrotti) [reply packages/icon/style.ts]: Woops, no

File: packages/icon/components/Icon.tsx (hunk 2 above)
Comment (mperrotti) [reply packages/icon/components/Icon.tsx]: I was having trouble with this one. String enums that use constants as an initializer are a big pain

File: packages/design-tokens/index.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
+
```
Comment (mperrotti) [inline packages/design-tokens/index.js]: I accidentally added this file. Will be removing

**Comment by mperrotti** [comment]
I have no idea why Jenkins can't get past the authorization step, and I have no option to restart it :/

FWIW: It was building without any issues earlier today

**Comment by GeorgiSTodorov** [comment]
I retriggered in the old Jenkins view, but it failed again.
screenshot from 2019-01-17 14-40-12

**Comment by mperrotti** [comment]
Closing this because I opened a new PR here: https://github.com/dcos-labs/ui-kit/pull/278

Jenkins was acting strange

**Pull request closed without merging**